projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_detection_2d_gem_node.py

```diff
@@ -0,0 +1,293 @@
+        )
+        self.gem_learner.fusion_method = "sc_avg"
+        self.gem_learner.download(path=".", verbose=True)
+
+        # Subscribers
+        msg_color = message_filters.Subscriber(self, ROS_Image, input_color_image_topic)
+        msg_ir = message_filters.Subscriber(self, ROS_Image, input_infra_image_topic)
+
+        sync = message_filters.TimeSynchronizer([msg_color, msg_ir], 1)
+        sync.registerCallback(self.callback)
+
+    def callback(self, msg_color, msg_ir):
+        """
+        Callback that process the input data and publishes to the corresponding topics
+        :param msg_color: input color image message
+        :type msg_color: sensor_msgs.msg.Image
+        :param msg_ir: input infrared image message
+        :type msg_ir: sensor_msgs.msg.Image
+        """
+        # Convert images to OpenDR standard
+        image_color = self.bridge.from_ros_image(msg_color).opencv()
+        image_ir_raw = self.bridge.from_ros_image(msg_ir, "bgr8").opencv()
+        image_ir = cv2.warpPerspective(image_ir_raw, self.h, (image_color.shape[1], image_color.shape[0]))
+
+        # Perform inference on images
+        boxes, w_sensor1, _ = self.gem_learner.infer(image_color, image_ir)
+
+        #  Annotate image and publish results:
+        if self.detection_publisher is not None:
+            ros_detection = self.bridge.to_ros_bounding_box_list(boxes)
+            self.detection_publisher.publish(ros_detection)
+            # We can get the data back using self.bridge.from_ros_bounding_box_list(ros_detection)
+            # e.g., opendr_detection = self.bridge.from_ros_bounding_box_list(ros_detection)
+
+        if self.color_publisher is not None:
+            plot_color = draw_bounding_boxes(image_color, boxes, class_names=self.classes)
+            message = self.bridge.to_ros_image(Image(np.uint8(plot_color)))
+            self.color_publisher.publish(message)
+        if self.ir_publisher is not None:
+            plot_ir = draw_bounding_boxes(image_ir, boxes, class_names=self.classes)
+            message = self.bridge.to_ros_image(Image(np.uint8(plot_ir)))
+            self.ir_publisher.publish(message)
+
+
+def main(args=None):
+    rclpy.init(args=args)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--input_color_image_topic", help="Topic name for input color image", type=str, default="/camera/color/image_raw"
+    )
+    parser.add_argument(
+        "--output_color_image_topic",
+        help="Topic name for output annotated color image",
+        type=str,
+        default="/opendr/color_objects_annotated",
+    )
+    parser.add_argument(
+        "--input_infra_image_topic", help="Topic name for input infra image", type=str, default="/camera/infra/image_rect_raw"
+    )
+    parser.add_argument(
+        "--output_infra_image_topic",
+        help="Topic name for output annotated infra image",
+        type=str,
+        default="/opendr/infra_objects_annotated",
+    )
```

Comment:
ROS1 node comment for "color"->"rgb" and default value for topics also applies here.